astropy/convolution/setup_package.py

```diff
@@ -1,12 +1,12 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
+import os
 import sys
-from pathlib import Path
 
 from numpy import get_include as get_numpy_include
 from setuptools import Extension
 
-C_CONVOLVE_PKGDIR = Path(__file__).parent.resolve().relative_to(Path.cwd())
+C_CONVOLVE_PKGDIR = os.path.relpath(os.path.dirname(__file__))
```

Comment:
```suggestion
C_CONVOLVE_PKGDIR = Path(os.path.relpath(os.path.dirname(__file__)))
```

Reply:
Then most other changes aren't necessary. As stuff improves we just systematically remove the strings manipulation in favor of file path manipulation.

Reply:
Ha, that could be it, I didn't know about this !

Reply:
Actually, do you have a source to back it up ? I couldn't find anything about this in https://docs.python.org/3/library/pathlib.html